spec/OpenLineage.md

```diff
@@ -0,0 +1,110 @@
+# OpenLineage Spec
+
+## Specification
+
+The specification for OpenLineage is formalized as an OpenAPI spec: [OpenLineage.yml](OpenLineage.yml)
+It allows extensions to the spec using `Custom Facets` as described in this document.
+
+## Core concepts
+
+### Core Lineage Model
+
+![Open Lineage model](OpenLineageModel.svg)
+
+- **Run State Update**: and event describing an observed state of a job run. It is required to at least send one event for a START transition and a COMPLETE/FAIL/ABORT transition. Aditional events are optional.
+
+- **Job**: a process definition that consumes and produces datasets (defined as its inputs and outputs). It is identified by a unique name within a namespace (which is typicaly assigned to the scheduler starting the jobs). The *Job* evolves over time and this change is captured when the job runs.
+
+- **Dataset**: an abstract representation of data. It has a unique name within a namespace derived from its physical location (for example db.host.database.schema.table). Typicaly, a *Dataset* changes when a job writing to it completes.
+
+- **Run**: An instance of a running job with a start and completion (or failure) time. It is uniquely identified by an id relative to its job definition.
+
+- **Facet**: A piece of metadata attached to one of the entities defined above.
+
+example:
+Here is an example of a simple start run event not adding any facet information:
+```
+{
+  "transition": "START",
+  "eventTime": "2020-12-09T23:37:31.081Z",
+  "run": {
+    "runId": "345",
+  },
+  "job": {
+    "namespace": "my-scheduler-namespace",
+    "name": "myjob.mytask",
+  },
+  "inputs": [
+    {
+      "namespace": "my-datasource-namespace",
+      "name": "instance.schema.table",
+    }
+  ],
+  "outputs": [
+    {
+      "namespace": "my-datasource-namespace",
+      "name": "instance.schema.output_table",
+    }
+  ],
+  "producer": "https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client"
+}
```

Comment:
I would expect the `job` to have `inputs` / `outputs` as fields, rather than be top-level fields associated with the run state change. That is:

```
job: {
  inputs: {...}
  outputs: {...}
}
```

But, if the goal is to associated the `inputs` / `outputs` at the run-level, then should `run` include them?

```
run: {
  inputs: {...}
  outputs: {...}
}
```